[#2] liveActivity 실행 로직 추가

leesungkug opened the pull request:
## 📝 작업 내용

1. liveActivity 생성 및 실행 로직 추가
2. liveActivity 업데이트 로직 추가
3. liveActivity 종료 로직 추가

leesungkug committed:
feat: [#2] Adding liveActivity execution logic

leesungkug committed:
feat: [#2] Adding liveActivity execution logic

leesungkug commented:
용량 조회하는 코드 제거했습니다.  capacityManager안에 함수처럼 LiveActivityManager를 사용하면 됩니다.

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+    @Published var freeSpace = ""
+    
+    var currentFreeSpace: String = ""
+    var isEnd = false
```
isEnded말고 End로 끝낸 이유가 있을까요?

leesungkug replied:
아하 수정하겠습니다!

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+    
+    func fetchDiskSpace() {
+        DispatchQueue.global().async {
+            if let spaceInfo = self.getDiskSpaceInfo() {
```
저의 GCD방식 말고 Concurrency 방식(ex. async, await)으로 하는 거 아니였나요 ..?

leesungkug replied:
이 부분은 임시 로직이여서 삭제되었습니다!

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+            let freeValues = try fileURL.resourceValues(forKeys: [.volumeAvailableCapacityForImportantUsageKey])
+            
+            
+            if let totalCapacity = totalValue.volumeTotalCapacity, let freeCapacity = freeValues.volumeAvailableCapacityForImportantUsage {
```
선언부가 너무 길어서.. 줄바꿈으로 나눠주면 좋을 거 같아요!

leesungkug replied:
넵 수정해놓겠습니다

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+     LiveActivity 생성
+     */
+    private func startLiveActivity() {
+        if #available(iOS 16.2, *) {
```
저희 최소 타겟 버전도 고려해보면 좋을 거 같아요!

leesungkug replied:
이 부분은 이야기 나눠봐야될 것 같아요

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+    private func startLiveActivity() {
+        if #available(iOS 16.2, *) {
+            if ActivityAuthorizationInfo().areActivitiesEnabled {
+                let activityData = dynamicCapacityAttributes(name: "남은용량")
```
해당 "남은용량" 부분을 영어로 표시는 불가능할까요?

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+    func fetchDiskSpace() {
+        DispatchQueue.global().async {
+            if let spaceInfo = self.getDiskSpaceInfo() {
+                DispatchQueue.main.async {
```
저같은 경우에는 manager 모델을 @mainActor로 선언을 했는데 이부분도 통일하면 좋을 거 같아요!

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,135 @@
+    private var currentID: String = ""
+    private var timer: Timer?
+    
+    init() {
```
만약 해당 코드를 singleton 패턴을 유지한다면, 해당 생성자에는 private 접근 제어 키워드를 붙여줘야 하는 거 아닌가요 ??
개인적인 생각으로 싱글톤을 굳이 써야할 이유가 없다면 SwiftUI에서는 EnvironmentObject가 더 맞다고 생각합니다!

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,83 @@
+import Combine
+import Foundation
+
+enum viewStatus {
```
코드가 짧더라도 타입 선언부를 따로 분리하는 것은 어떨까요?

zaehorang commented on MoonCrystal/MoonCrystal/CapacityManager.swift:
```diff
@@ -0,0 +1,83 @@
+    private var timer: Timer?
+    
+    /**
+     LiveActivity 스케쥴링 시작
```
CapacityManager 부분인데 LiveActivity 관련 메서드를 함께 쓰신 이유가 있을까요?
LiveActivity 관련 코드는 로직을 분리하는게 어떤가요 ?

zaehorang commented on MoonCrystal/MoonCrystal/DynamicCapacityAttributes.swift:
```diff
@@ -0,0 +1,18 @@
+
+struct dynamicCapacityAttributes: ActivityAttributes {
+    
+    public struct ContentState: Codable, Hashable {
```
해당 선언부에만 public을 붙인 이유가 있을까요??

zaehorang commented on MoonCrystal/MoonCrystal/LiveActivityManager.swift:
```diff
@@ -0,0 +1,74 @@
+@available(iOS 16.1, *)
+class LiveActivityManager{
+    /**
+     LiveActivity 생성 요청
```
여러 줄의 주석 처리가 아닌데 /** ~~ */ 로 표시한 이유가 있을까요?

zaehorang commented on MoonCrystal/MoonCrystal/LiveActivityManager.swift:
```diff
@@ -0,0 +1,74 @@
+import Foundation
+
+@available(iOS 16.1, *)
+class LiveActivityManager{
```
LiveActivityManager (여기 한 칸 띄우기!!) { }

zaehorang commented on MoonCrystal/MoonCrystal/LiveActivityManager.swift:
```diff
@@ -0,0 +1,74 @@
+    /**
+     LiveActivity 생성 요청
+     */
+    static func startLiveActivity(for activityData: dynamicCapacityAttributes, state contentState: dynamicCapacityAttributes.ContentState) -> String{
```
아래도 마찬가지인데...중괄호 쓸 때 양 옆 띄어주세요!
ex. -> String {
ex. do {
ex. } catch {

zaehorang replied:
해당 클래스의 메서드들을 모두 static으로 선언한 이유가 있나요?

zaehorang commented on MoonCrystal/MoonCrystalWidgetExtention/AppIntent.swift:
```diff
@@ -0,0 +1,18 @@
+//  MoonCrystalWidgetExtention
+//
+//  Created by sungkug_apple_developer_ac on 7/26/24.
+//
```
아직 dynamic island와 ActivityKit 부분을 잘 몰라서.. 좀 더 공부하고 나중에 같이 더 봐요! 🐯
but, 위에서 언급했던 내용들을 함께 수정부탁드려🐲

zaehorang commented on MoonCrystal/MoonCrystalWidgetExtention/MoonCrystalWidgetExtention.swift:
```diff
@@ -0,0 +1,84 @@
+}
+
+extension ConfigurationAppIntent {
+    fileprivate static var smiley: ConfigurationAppIntent {
```
잘 모르지만... fileprivate static으로 선언하는게 필수인건가요..?

zaehorang reviewed (commented):
코멘트 남겼습니다 !!🐯

Seon-U committed:
feat: [#3] Add MediaCapacityConverter

용량정보를 사진과 비디오타입으로 변환하는 함수들의 모음인 class와 비디오 포맷별 1분당 바이트 단위, 사진 포맷 당 1장의 바이트 단위를 반환하는 enum을 생성했습니다. 다만 구조화하면서 데이터매니저를 생성할 정도인가 의문이 들어 statc func 선언만 해둔 상태입니다.

leesungkug committed:
fix: [#2] Resolving the Info file error

leesungkug committed:
fix: [#2] Resolving the Info file error

Seon-U committed:
feat: [#3] Edit MediaCapacityConverter

- 앱 플로우에 맞춰 사진용량 기준을 default로 설정
- 변환된 값인 분단위 시간을 시간과 분으로 나눠 반납하는 Int extension 추가
- byte를 GB 단위로 바꾸는 Int extension 추가

Seon-U committed:
chore: [#3] Add uncommited change

- 이전 커밋에 올라가지 못했던 변화를 재 커밋함
- 앱 플로우에 맞춰 촬영 가능 사진값 계산을 default 기준으로 간소화
- Int+Extension에 주석을 파일명에 맞춰 수정하고 바이트를 GB로 바꾸는 extension 추가

Seon-U committed:
style: [#3] change code style

-enum VideoFormatCapacity를 별도 파일로 분리
-설명 주석을 수정
-func 파라미터명과 유형 사이 ":" 위치를 수정

Seon-U committed:
style: [#3] change code style

- 프레임워크 선언 사이 공백 수정

Seon-U committed:
feat: [#9] Add CapacityCalculator

-device의 전체 용량과 여유 용량을 계산하여 변수에 저장하는 class 생성

Seon-U committed:
style: [#9] Change data type of Storage

-Int64기준으로 반환 및 선언하던 값을 Int타입으로 변경

Seon-U committed:
Style  [#9]

- Int 선언부 제거

Seon-U committed:
Fix[#9] Add ObservableObject

-옵저버블 추가

Seon-U committed:
style: [#3] change annotation and spacing

-변수명 뒤에 타입 선언할 때 붙여쓰기로 수정
-생성시 기본 주석을 파일명에 맞춰 수정

Seon-U committed:
Style: [#3] delete unused framework

-Foundation을 선언하지 않아도 되는 화면에 있던 Foundation 제거

Seon-U committed:
style: [#3] change func name

- minuteToWholeTime -> minutesToHoursAndMinutes로 바꿈

Seon-U committed:
Style: [#9] change function name and value name

-totalStorage -> totalCapacity로 바꿨습니다
-checkDeviceStorage를 변수명과 일치하며 updateTotalCapacity로 바꿨습니다.

Seon-U committed:
Fix: [#9] fix function error

- updateTotalCapacity와 updateFreeCapacity 양쪽 모두 마지막에 0으로 초기화해두던 부분 삭제
- final, private 등 구체적으로 정해두고 효율화

leesungkug committed:
chore: [#2] Rebase Develop branch

leesungkug committed:
remove: [#2] Delete unnecessary file paths

leesungkug committed:
Merge branch 'develop' into feat/dynamic-island

zaehorang reviewed (commented):
고생했삼겹살

zaehorang reviewed (approved):
고생했삼삼삼
